### PR TITLE
[autorestart] Ignore known autorestart error logs and fix all_critical_process_status bug in sonic.py

### PR DESCRIPTION
**Pull request by lerry-lee**
### Description of PR

Update expected error logs by ignoring the case of container's first letter. Ignore known syncd error logs in 202205 and 202012
Fix all_critical_process_status bug in sonic.py when parsing service critical processes.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205

#### What is the motivation for this PR?

While testing service container auto-restart, loganalyzer analyzes some error logs and reports them as errors, sometimes these error logs are known and can be ignored. And in `syslog` some service containers failed to start and its log like 'ERR systemd...Failed to start swss container', but mux is like 'ERR systemd...Failed to start mux Container'. Its only defference is the word 'container''s first letter case.

In `tests/common/devices/sonic.py`'s `all_critical_process_status` method, when parsing service containers' critical processes, if the result of command execution is not right, the service status may be wrong.
Some logs are below:
```
        {
            "stderr_lines": [
                "Error response from daemon: Container 3a9dc286fe03c968e60cd89db7431745c0901111d4c32a0e0eeb52a5f5cd12d9 is not running"
            ],
            "stderr": "Error response from daemon: Container 3a9dc286fe03c968e60cd89db7431745c0901111d4c32a0e0eeb52a5f5cd12d9 is not running\n",
            "stdout": "",
            "stdout_lines": [],
            "cmd": "docker exec swss supervisorctl status",
            "rc": 1
        }
```

```
        {
            "stderr_lines": [],
            "stderr": "",
            "stdout": "error: <class 'xmlrpc.client.Fault'>, <Fault 6: 'SHUTDOWN_STATE'>: file: /usr/lib/python3.9/xmlrpc/client.py line: 662\n",
            "stdout_lines": [
                "error: <class 'xmlrpc.client.Fault'>, <Fault 6: 'SHUTDOWN_STATE'>: file: /usr/lib/python3.9/xmlrpc/client.py line: 662"
            ],
            "cmd": "docker exec teamd supervisorctl status",
            "rc": 1
        }
```

```
        {
            "stderr_lines": [],
            "stderr": "",
            "stdout": "unix:///var/run/supervisor.sock no such file\n",
            "stdout_lines": [
                "unix:///var/run/supervisor.sock no such file"
            ],
            "cmd": "docker exec pmon supervisorctl status",
            "rc": 4
        }
```

In these cases, the service status should be false because the container is not running healthy, but will be dealt with true currently.

#### How did you do it?

Ignore known syncd error logs and update expected error log regular expression for failed to start issue.

When parsing the result of command "docker exec <container_name> supervisorctl status", if stdout_lines is empty or is error message, the service container's status should be false.

#### How did you verify/test it?

Run TC in dual-tor and single-tor testbed, as well as SONiC 202020 and 202205.

### CODE REVIEW

File: tests/autorestart/test_container_autorestart.py [hunk 1]
```diff
@@ -75,6 +75,19 @@ def ignore_expected_loganalyzer_exception(duthosts, enum_rand_one_per_hwsku_host
...
+            ".*ERR syncd[0-9]*#SDK.*mlnx_bridge_1d_oid_to_data: Unexpected bridge type 0 is not 1D.*",
+            ".*ERR syncd[0-9]*#SDK.*mlnx_bridge_port_lag_or_port_get: Invalid port type - 2.*",
+            ".*ERR syncd[0-9]*#SDK.*mlnx_bridge_port_isolation_group_get: Isolation group is only supported for bridge port type port.*",
+            ".*ERR syncd[0-9]*#SDK.*mlnx_debug_counter_availability_get: Unsupported debug counter type - (0|1).*",
+            ".*ERR syncd[0-9]*#SDK.*mlnx_get_port_stats_ext: Invalid port counter (177|178|179|180|181|182).*",
+            ".*ERR syncd[0-9]*#SDK.*Failed getting attrib SAI_BRIDGE_ATTR_BROADCAST_FLOOD_GROUP.*",
+            ".*ERR syncd[0-9]*#SDK.*Failed getting attrib SAI_BRIDGE_ATTR_UNKNOWN_MULTICAST_FLOOD_GROUP.*",
+            ".*ERR syncd[0-9]*#SDK.*Failed getting attrib SAI_BRIDGE_ATTR_UNKNOWN_UNICAST_FLOOD_GROUP.*",
+            ".*ERR syncd[0-9]*#SDK.*Failed getting attrib SAI_BRIDGE_PORT_ATTR_ISOLATION_GROUP.*",
+            ".*ERR syncd[0-9]*#SDK.*Failed getting attrib SAI_BRIDGE_PORT_ATTR_PORT_ID.*",
```
Comment (wangxin) [inline tests/autorestart/test_container_autorestart.py:88]: Patterns from line 84-88 may can be combined into a single one. For example:
```
".*ERR syncd[0-9]*#SDK.*Failed getting attrib SAI_BRIDGE_.*",
```

File: tests/common/devices/sonic.py [hunk 2]
```diff
@@ -642,12 +642,21 @@ def all_critical_process_status(self):
...
                 pname, status, _ = re.split('\s+', line, 2)
                 if status != 'RUNNING':
                     if pname in service_group_process['groups'] or pname in service_group_process['processes']:
                         service_critical_process['exited_critical_process'].append(pname)
                         service_critical_process['status'] = False
+                    # Sometimes, stdout_lines may be error messages but not emtpy
+                    # In this situation, service container status should be false
+                    # We can check status is valid or not
+                    if status not in ('EXITED'):
+                        service_critical_process['status'] = False
```
Comment (wangxin) [inline tests/common/devices/sonic.py:659]: This is a little bit confusing. Can you improve the logic here?

File: tests/autorestart/test_container_autorestart.py [hunk 3]
```diff
@@ -88,7 +101,7 @@ def ignore_expected_loganalyzer_exception(duthosts, enum_rand_one_per_hwsku_host
     ignore_regex_dict = {
         'common' : [
             ".*ERR monit.*",
-            ".*ERR systemd.*Failed to start .* container*",
+            ".*ERR systemd.*Failed to start .* (C|c)ontainer.*",
```
Comment (wangxin) [inline tests/autorestart/test_container_autorestart.py:104]: The regular expression of `(C|c)` can be `[Cc]`.

File: tests/common/devices/sonic.py (hunk 2 above)
Comment (lerry-lee) [reply tests/common/devices/sonic.py]: Thanks, I will improve it.

File: tests/autorestart/test_container_autorestart.py (hunk 1 above)
Comment (lerry-lee) [reply tests/autorestart/test_container_autorestart.py]: Good suggestion, thanks.

File: tests/common/devices/sonic.py [hunk 4]
```diff
@@ -642,9 +642,18 @@ def all_critical_process_status(self):
...
+                service_critical_process['status'] = False
             for line in service_result['stdout_lines']:
                 pname, status, _ = re.split('\s+', line, 2)
-                if status != 'RUNNING':
+                # Sometimes, stdout_lines may be error messages but not emtpy
+                # In this situation, service container status should be false
+                # We can check status is valid or not
+                if status not in ('RUNNING','EXITED','STOPPED'):
+                    service_critical_process['status'] = False
+                elif status != 'RUNNING':
```
Comment (wangxin) [inline tests/common/devices/sonic.py:656]: This logic is much better!

**Review by wangxin — APPROVED** [review]

**Pull request merged**